src/Stratis.Bitcoin.Features.BlockStore/BlockStoreCache.cs

```diff
@@ -90,69 +118,31 @@ public void Expire(uint256 blockid)
             block = await this.blockRepository.GetAsync(blockid);
             if (block != null)
             {
-                this.cache.Set(blockid, block, TimeSpan.FromMinutes(10));
+                this.cache.Set(blockid, block, this.blockEntryOptions);
                 this.PerformanceCounter.AddCacheSetCount(1);
             }
 
             this.logger.LogTrace("(-)[CACHE_MISS]:'{0}'", block);
             return block;
         }
 
-        public async Task<Block> GetBlockByTrxAsync(uint256 trxid)
-        {
-            this.logger.LogTrace("({0}:'{1}')", nameof(trxid), trxid);
-            Guard.NotNull(trxid, nameof(trxid));
-
-            uint256 blockid;
-            Block block;
-            if (this.cache.TryGetValue(trxid, out blockid))
-            {
-                this.PerformanceCounter.AddCacheHitCount(1);
-                block = await this.GetBlockAsync(blockid);
-                return block;
-            }
-
-            this.PerformanceCounter.AddCacheMissCount(1);
-
-            blockid = await this.blockRepository.GetTrxBlockIdAsync(trxid);
-            if (blockid == null)
-            {
-                this.logger.LogTrace("(-):null");
-                return null;
-            }
-
-            this.cache.Set(trxid, blockid, TimeSpan.FromMinutes(10));
-            this.PerformanceCounter.AddCacheSetCount(1);
-            block = await this.GetBlockAsync(blockid);
-
-            this.logger.LogTrace("(-):'{0}'", block);
-            return block;
-        }
-
-        public async Task<Transaction> GetTrxAsync(uint256 trxid)
-        {
-            this.logger.LogTrace("({0}:'{1}')", nameof(trxid), trxid);
-            Guard.NotNull(trxid, nameof(trxid));
-
-            Block block = await this.GetBlockByTrxAsync(trxid);
-            Transaction trx = block?.Transactions.Find(t => t.GetHash() == trxid);
-
-            this.logger.LogTrace("(-):'{0}')", trx);
-            return trx;
-        }
-
         public void AddToCache(Block block)
         {
             uint256 blockid = block.GetHash();
             this.logger.LogTrace("({0}:'{1}')", nameof(block), blockid);
 
-            Block existingBlock;
-            if (!this.cache.TryGetValue(blockid, out existingBlock))
-                this.cache.Set(blockid, block, TimeSpan.FromMinutes(10));
+            if (!this.cache.TryGetValue(blockid, out Block existingBlock))
+                this.cache.Set(blockid, block, this.blockEntryOptions);
 
             this.logger.LogTrace("(-)[{0}]", existingBlock != null ? "ALREADY_IN_CACHE" : "ADDED_TO_CACHE");
         }
 
+        /// <inheritdoc />
+        public bool Exist(uint256 blockid)
```

Comment:
We dont really use this do we?

Reply:
Nope, just added this because we might need it. Can remove that as well.

Reply:
no thats fine